lightning/src/ln/channelmanager.rs

```diff
@@ -3468,40 +3472,60 @@ where
 
 		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
-		let (chan, msg) = match peer_state.outbound_v1_channel_by_id.remove(&temporary_channel_id) {
-			Some(chan) => {
-				let funding_txo = find_funding_output(&chan, &funding_transaction)?;
-
-				let funding_res = chan.get_funding_created(funding_transaction, funding_txo, &self.logger)
-					.map_err(|(mut chan, e)| if let ChannelError::Close(msg) = e {
-						let channel_id = chan.context.channel_id();
-						let user_id = chan.context.get_user_id();
-						let shutdown_res = chan.context.force_shutdown(false);
-						let channel_capacity = chan.context.get_value_satoshis();
-						(chan, MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, user_id, shutdown_res, None, channel_capacity))
-					} else { unreachable!(); });
-				match funding_res {
-					Ok((chan, funding_msg)) => (chan, funding_msg),
-					Err((chan, err)) => {
-						mem::drop(peer_state_lock);
-						mem::drop(per_peer_state);
-
-						let _: Result<(), _> = handle_error!(self, Err(err), chan.context.get_counterparty_node_id());
-						return Err(APIError::ChannelUnavailable {
-							err: "Signer refused to sign the initial commitment transaction".to_owned()
-						});
-					},
+		match peer_state.channel_by_id.entry(*temporary_channel_id) {
+			hash_map::Entry::Occupied(chan_phase_entry) => {
+				if let ChannelPhase::UnfundedOutboundV1(_) = chan_phase_entry.get() {
+					// We need to take ownership of the entry and promote the unfunded channel to a `Channel`,
```

Comment:
Yeah that was only one lookup. There's no way in that case to avoid the removal and re-add of a channel of different type. I wish there was.

But 1 lookup vs 2 in happy path seems like a good enough reason to revert. So I'll do so.